test/e2e/rc.go

```diff
@@ -71,63 +95,46 @@ func TestBasicImage(c *client.Client, test string, image string) bool {
 			},
 		},
 	})
-	if err != nil {
-		glog.Infof("Failed to create replication controller %s: %v", name, err)
-		return false
-	}
+	Expect(err).NotTo(HaveOccurred())
 	// Cleanup the replication controller when we are done.
 	defer func() {
 		// Resize the replication controller to zero to get rid of pods.
 		controller.Spec.Replicas = 0
 		if _, err = c.ReplicationControllers(ns).Update(controller); err != nil {
-			glog.Errorf("Failed to resize replication controller %s to zero: %v", name, err)
+			Logf("Failed to resize replication controller %s to zero: %v", name, err)
 		}
 
 		// Delete the replication controller.
 		if err = c.ReplicationControllers(ns).Delete(name); err != nil {
-			glog.Errorf("Failed to delete replication controller %s: %v", name, err)
+			Logf("Failed to delete replication controller %s: %v", name, err)
 		}
-
-		// Report running time for test.
-		glog.Infof("Controller %s test took %v seconds", name, time.Since(testStart).Seconds())
 	}()
 
 	// List the pods, making sure we observe all the replicas.
 	listTimeout := time.Minute
 	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
 	pods, err := c.Pods(ns).List(label)
-	if err != nil {
-		glog.Errorf("Controller %s: Failed to list pods: %v", name, err)
-	}
+	Expect(err).NotTo(HaveOccurred())
 	t := time.Now()
 	for {
```

Comment:
Yeah, I noticed this as I was porting it. We now have three timeout mechanisms that I know of in the system There's this one, which is actually nice because it's "true". There's this one: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/util.go#L43 (which uses the "sleep and add up the sleeps" method). And then when I was porting Brendan's endpoints test, I gutted his goroutine based individual timeouts and actually forced the test into a full test timeout: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/service.go#L248 (which is the combination of that timeout and the "done Done" on the function signature).

I can't decide if we want fine-grained timeouts for individual actions or to just go through and profile whole tests, honestly. We probably want some combination of both.